File: reduxCore/src/main/java/com/develop/zuzik/redux/core/store/Middleware.java
package com.develop.zuzik.redux.core.store;

import org.jetbrains.annotations.NotNull;

import io.reactivex.Observable;

/**
 * Created by yaroslavzozulia on 9/25/17.
 */

public interface Middleware<State> {
	@NotNull
	Observable<State> dispatch(@NotNull Observable<State> state, @NotNull Action action);
}

File: reduxCore/src/main/java/com/develop/zuzik/redux/core/Middleware.kt
package com.develop.zuzik.redux.core

import io.reactivex.Observable

/**
 * Created by yaroslavzozulia on 9/24/17.
 */
//TODO: use it like lambda
interface Middleware<State> {
	fun dispatch(state: Observable<State>, action: Action): Observable<State>
}